pkg/gofr/datasource/opentsdb/response.go

```diff
@@ -0,0 +1,416 @@
+	jsonBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		errReading := fmt.Errorf("failed to read response body for %s %s: %w", method, url, err)
+
+		message = fmt.Sprint(errReading)
+
+		return errReading
+	}
+
+	parser := parsedResp.getCustomParser(ctx, c.logger, c.tracer)
+	if parser == nil {
+		// Use the default JSON unmarshaller if no custom parser is provided.
+		if err = json.Unmarshal(jsonBytes, parsedResp); err != nil {
+			errUnmarshaling := fmt.Errorf("failed to unmarshal response body for %s %s: %w", method, url, err)
+
+			message = fmt.Sprint(errUnmarshaling)
+
+			return errUnmarshaling
+		}
+	} else {
+		// Use the custom parser if available.
+		if err := parser(jsonBytes); err != nil {
+			message = fmt.Sprintf("failed to parse response body through custom parser %s %s: %v", method, url, err)
+			return err
+		}
+	}
+
+	status = statusSuccess
+	message = fmt.Sprintf("%s request sent at : %s", method, url)
+
+	return nil
+}
+
+func (c *Client) version(ctx context.Context, verResp *VersionResponse) error {
+	span := c.addTrace(ctx, "Version")
+
+	status := statusFailed
+
+	var message string
+
+	defer sendOperationStats(c.logger, time.Now(), "Version", &status, &message, span)
+
+	verEndpoint := fmt.Sprintf("%s%s", c.endpoint, versionPath)
+
+	if err := c.sendRequest(ctx, http.MethodGet, verEndpoint, "", verResp); err != nil {
+		message = fmt.Sprintf("error while processing request at URL %s: %s", verEndpoint, err)
+		return err
+	}
+
+	status = statusSuccess
+	message = "version response retrieved successfully."
+
+	return nil
+}
+
+// isValidOperateMethod checks if the provided HTTP method is valid for
+// operations such as POST, PUT, or DELETE.
+func (c *Client) isValidOperateMethod(ctx context.Context, method string) bool {
+	span := c.addTrace(ctx, "isValidOperateMethod")
+
+	status := statusSuccess
+
+	var message string
+
+	defer sendOperationStats(c.logger, time.Now(), "isValidOperateMethod", &status, &message, span)
+
+	method = strings.TrimSpace(strings.ToUpper(method))
+	if method == "" {
+		return false
+	}
+
+	validMethods := []string{http.MethodPost, http.MethodPut, http.MethodDelete, http.MethodPut}
+	for _, validMethod := range validMethods {
+		if method == validMethod {
+			return true
+		}
+	}
+
+	return false
+}
+
+func customParserHelper(ctx context.Context, resp genericResponse, operation string, logger Logger, tracer trace.Tracer,
+	unmarshalFunc func([]byte) error) func([]byte) error {
+	span := resp.addTrace(ctx, tracer, operation)
+
+	status := statusFailed
+
+	var message string
+
+	defer sendOperationStats(logger, time.Now(), operation, &status, &message, span)
+
+	return func(result []byte) error {
+		err := unmarshalFunc(result)
+		if err != nil {
+			message = fmt.Sprintf("unmarshal %s error: %s", operation, err)
+			logger.Errorf(message)
+
+			return err
+		}
+
+		status = statusSuccess
+		message = fmt.Sprintf("%s custom parsing was successful.", operation)
```

Comment:
I don't think other messages have a leading dot

```suggestion
		message = fmt.Sprintf("%s custom parsing was successful", operation)
```